src/nullablevector.jl

```diff
@@ -311,3 +311,80 @@ function Base.empty!(X::NullableVector)
     empty!(X.isnull)
     return X
 end
+
+function Base.promote_rule{T1,T2}(::Type{T1}, ::Type{Nullable{T2}})
```

Comment:
Yes, you're right. I see now that `promote_type` has the consistency I was expecting from `promote_rule`. I'll need to learn more about those differences.

```julia
julia> promote_rule(Nullable{Int64}, Int64)
Nullable{Int64}

julia> promote_rule(Int64, Nullable{Int64})
Union{}

julia> promote_type(Nullable{Int64}, Int64)
Nullable{Int64}

julia> promote_type(Int64, Nullable{Int64})
Nullable{Int64}
```